### PR TITLE
Reduce CI go module downloads, add make targets

### PR DESCRIPTION
**Pull request by silverwind**
The CI currently downloads all go modules in each pipeline step because go modules reside outside the project directory. Fix this by introducing a volume for the `/go` directory [1] so modules are only downloaded once per pipeline using a new `deps-backend` make target.

For completeness, I also included new `deps` and `deps-frontend` targets and the frontend one is also triggered explicitly on CI where needed.

[1] https://docs.drone.io/pipeline/kubernetes/examples/language/golang/#dependencies

### CODE REVIEW

**Comment by silverwind** [comment]
Seems there is a race condition in the lint steps which all try to install `golangci-lint` in parallel into the shared volume, resulting in `golangci-lint: Text file busy`. Probably need to extract tool installation to a separate step so it only runs once.

**Comment by silverwind** [comment]
Don't think it need privileges.

**Comment by silverwind** [comment]
Few more downloads to eliminate, on average, this saves about 15 seconds per step that previously downloaded modules.

**Comment by techknowlogick** [comment]
Awesome, thanks :)

I've restarted the build

**Comment by silverwind** [comment]
Should be ready for review.

There is one unrelated change included here which may speed up `golangci-lint` a bit by skipping over certain directories (thought it will not work for all linters, see https://github.com/golangci/golangci-lint/issues/913.

File: .drone.yml [hunk 1]
```diff
@@ -27,17 +40,16 @@ steps:
     depends_on: [deps-frontend]
 
   - name: lint-backend
-    pull: always
```
Comment (lunny) [inline .drone.yml:30]: Why remove this?

Comment (silverwind) [reply .drone.yml:30]: `pull: always` is only necessary on the first occurence of an image in the run, subsequent pulls do not result in any new data, so they are pointless.

Comment (silverwind) [reply .drone.yml:30]: Actually, this needs to be added again because it is the first occurence. Will fix.

**Comment by silverwind** [comment]
@lunny I've gone through all `pull` statements and fixed them so they always are used on the first occurence of an image in a pipeline and the order in yaml is now `image` followed by `pull` which makes more sense to me.

**Review by techknowlogick — APPROVED** [review]

**Review by lafriks — APPROVED** [review]

**Comment by zeripath** [comment]
make lgtm work

**Comment by zeripath** [comment]
I think we might want to consider backporting this to 1.16.

**Comment by silverwind** [comment]
Right, there's no reason not to.

**Pull request merged**